### DISCUSSION TITLE
Joining Existing Thread Network

### DISCUSSION THREAD

**Discussion by modernoutlaw** [openthread/openthread#8367] [Q&A]
Hi! I have an existing Thread network I am trying to join using a Nordic dongle as an RCP. I cannot get the RCP to 'attach' and the WPAN service is 'associating'. Attached is the otbr-agent log. I am missing something but I can't figure out what it is and I am new to Thread. Any help is most appreciated!
<img width="1126" alt="Screen Shot 2022-11-05 at 9 56 31 AM" src="https://user-images.githubusercontent.com/117526432/200126106-77484859-b38a-43db-baaf-3a04c0671b70.png">

**Comment by jwhui** [comment]
Can you provide more detail on how you are trying to join the Thread device? What CLI commands and/or APIs are you calling?

You may find the [Thread Border Router codelab](https://openthread.io/codelabs/openthread-border-router#3) useful. It contains a section on joining a device to an existing Thread network.

**Reply by modernoutlaw** [reply]
Hi Jonathan! I have an nrf82540 dongle that I flashed with the open thread sample config from Githud. Then, I ran an OpenThread Border Router docker image using CLI from Nordic. The RCP will not move to the 'attached' state despite my efforts.

**Reply by jwhui** [reply]
What CLI commands and/or APIs are you calling? How exactly are you configuring the Thread network credentials?

**Reply by modernoutlaw** [reply]
Once the docker container is up and running I'm only attempting to join it to the Thread network via the OpenThread web portal using the Join button. I have not specified any network credentials, only the Prefix, and the Network Key which is the 0011223344...ff value. Maybe this is where I am missing something? Thanks!

**Reply by jwhui** [reply]
How did you set up your existing Thread network? Do you know it's network key?

**Reply by modernoutlaw** [reply]
It is a Thread network with my AppleTV and a few Eve Home devices. I set it up via the Eve app and HomeKit. I do not know it's Network Key. Is there any way to figure that out?

**Reply by jwhui** [reply]
Thread networks have security enabled by default. Possession of the network key is required in order to communicate with it. Unfortunately, I do not know of an easy way currently to extract the network key via homekit devices.

**Reply by modernoutlaw** [reply]
Understood, thank you Jonathan for your help and understanding!
